decavision/model_training/tfrecords_image_classifier.py

```diff
@@ -298,21 +298,32 @@ def fit(self, save_model=None, export_model=None, min_accuracy=None,
             verbose (bool): show details of training or not
             fine_tuning (bool): fine tune pretrained model or not
             l2_lambda (float): amount of L2 regularization to include in extra layer
-            min_accuracy (float): if specified, stop training when improvement in val accuracy is smaller than min_accuracy
+            patience (int): if specified, stops training when improvement in val accuracy is not observed using early stopping
             logs (str): if specified, tensorboard is used and logs are saved at this location
         """
         callbacks = None
+        # use reduce learning rate and early stopping callbacks
+        reduce_lr = tf.keras.callbacks.ReduceLROnPlateau(monitor='val_sparse_categorical_accuracy',
+                                                         factor=0.1,
+                                                         patience=5,
+                                                         mode='max')
+        callbacks = [reduce_lr]
+
         if logs:
             logdir = os.path.join(
                 logs, datetime.datetime.now().strftime("%Y%m%d-%H%M%S"))
             print('Fit log dir : ' + logdir)
             tensorboard_callback = tf.keras.callbacks.TensorBoard(logdir)
-            callbacks = [tensorboard_callback]
+            if not callbacks:
```

Comment:
Since callbacks is always defined to include reduce_lr, I think you only need to append tensorboard here. The if is not necessary.